### ISSUE TITLE
missing rpath in generated spdlog.pc file

### ISSUE THREAD

**Issue by attugit** [gabime/spdlog#1254]
generated spdlog.pc file on my system is:
```
# cat /usr/lib/pkgconfig/spdlog.pc
prefix=/usr
exec_prefix=${prefix}
includedir=${prefix}/include
libdir=${exec_prefix}/lib

Name: libspdlog
Description: Fast C++ logging library.
URL: https://github.com/gabime/spdlog
Version: 1.4.1
CFlags: -I${includedir}/spdlog -DSPDLOG_FMT_EXTERNAL
Libs: -L${libdir}/spdlog -lspdlog
```
Shared library is placed in own subdirectory under `/usr/lib`:
`/usr/lib/spdlog/libspdlog.so.1.4.1`

In the `Libs` section the capital `-L` helps to find it, but only during compile time (actually during linking).
When a binary is executed and there is no LD_LIBRARY_PATH pointing to this `/usr/lib/spdlog/` directory ldd complains.It can be fixed by adding a `rpath` or moving shared object 
`/usr/lib/libspdlog.so.1.4.1`

Secondary thing is that spdlog depends on fmt, this in most systems should be visable on a pkg level, but it would be also nice to put explicit `Requires: fmt` to pc file.

altogether I had to patch spdlog.pc file generated on my system with the following diff:
```
--- spdlog.pc.orig	2019-10-02 17:37:50.393079946 +0200
+++ spdlog.pc	2019-10-02 17:37:03.059747157 +0200
@@ -9,0 +10 @@
+Requires: fmt
@@ -11 +12 @@
-Libs: -L${libdir}/spdlog -lspdlog
+Libs: -Wl,-rpath,${libdir}/spdlog -L${libdir}/spdlog -lspdlog
``

**Comment by gabime** [comment]
>Shared library is placed in own subdirectory under /usr/lib:
/usr/lib/spdlog/libspdlog.so.1.4.1

Thanks for reporting - this was already resolved lately (#1253)

As for "Requires" - need to check how to add it to CMake, but only if SPDLOG_FMT_EXTERNAL is ON

**Comment by gabime** [comment]
Fixed is cf2bf488a22373b46eee36513335ff0e44355abc
